omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,253 @@
+		
+	}
+
+	@Test
+	public void shouldSaveASequentialIdentifierGenerator() throws Exception {
+		
+		long initialIdentifierSourceCount = getAllCount();
+		
+		SimpleObject identifierSource = new SimpleObject();
+		identifierSource.add("name", "test sequential identifier");
+		identifierSource.add("description", "test sequential identifier");
+		identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+		identifierSource.add("firstIdentifierBase", "1");
+		identifierSource.add("prefix", "ABC");
+		identifierSource.add("suffix", "XYZ");
+		identifierSource.add("minLength", "36");
+		identifierSource.add("maxLength", "38");
+		identifierSource.add("baseCharacterSet", "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789");
+		identifierSource.add("sourceType", SEQUENTIAL_IDENTIFIER_SOURCE_TYPE);
+		
+		String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+				
+		MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+				
+		handle(req);
+		
+		SequentialIdentifierGenerator generatedIdentifierSource = (SequentialIdentifierGenerator) service.getAllIdentifierSources(false).get(7);
+				
+		assertEquals(generatedIdentifierSource.getName(), "test sequential identifier");
+		assertEquals(generatedIdentifierSource.getDescription(), "test sequential identifier");
+		assertEquals(generatedIdentifierSource.getFirstIdentifierBase(), "1");
+		assertEquals(generatedIdentifierSource.getPrefix(), "ABC");
+		assertEquals(generatedIdentifierSource.getSuffix(), "XYZ");
+		assertEquals(generatedIdentifierSource.getMinLength().toString(), "36");
+		assertEquals(generatedIdentifierSource.getMaxLength().toString(), "38");
+		assertEquals(generatedIdentifierSource.getBaseCharacterSet(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789");
+		assertEquals(generatedIdentifierSource.getIdentifierType().getUuid(), IDENTIFIER_TYPE_UUID);
+		assertEquals(initialIdentifierSourceCount + 1, getAllCount());
+		
+	}
+	
+	@Test
+	public void shouldSaveARemoteIdentifierGenerator() throws Exception {
+		
+		long initialIdentifierSourceCount = getAllCount();
+		
+		SimpleObject identifierSource = new SimpleObject();
+		identifierSource.add("name", "test remote identifier");
+		identifierSource.add("description", "test remote identifier");
+		identifierSource.add("url", "http://localhost");
+		identifierSource.add("username", "openmrsuser");
+		identifierSource.add("password", "openmrspassword");
+		identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+		identifierSource.add("sourceType", REMOTE_IDENTIFIER_SOURCE_TYPE);
+		
+		String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+				
+		MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+				
+		handle(req);
+		
+		RemoteIdentifierSource generatedIdentifierSource = (RemoteIdentifierSource) service.getAllIdentifierSources(false).get(7);
```

Comment:
Instead of fetching all, what do you use getIdentifierSourceByUuid()?

Reply:
fixed